examples/consistency_models/train_consistency_distillation.py

```diff
@@ -0,0 +1,724 @@
+    if accelerator.is_local_main_process:
+        datasets.utils.logging.set_verbosity_warning()
+        diffusers.utils.logging.set_verbosity_info()
+    else:
+        datasets.utils.logging.set_verbosity_error()
+        diffusers.utils.logging.set_verbosity_error()
+
+    # Handle the repository creation
+    if accelerator.is_main_process:
+        if args.push_to_hub:
+            if args.hub_model_id is None:
+                repo_name = get_full_repo_name(Path(args.output_dir).name, token=args.hub_token)
+            else:
+                repo_name = args.hub_model_id
+            create_repo(repo_name, exist_ok=True, token=args.hub_token)
+            repo = Repository(args.output_dir, clone_from=repo_name, token=args.hub_token)
+
+            with open(os.path.join(args.output_dir, ".gitignore"), "w+") as gitignore:
+                if "step_*" not in gitignore:
+                    gitignore.write("step_*\n")
+                if "epoch_*" not in gitignore:
+                    gitignore.write("epoch_*\n")
+        elif args.output_dir is not None:
+            os.makedirs(args.output_dir, exist_ok=True)
+
+    # Initialize the model, using a smaller model than the one defined in the original paper by default
+    if args.model_config_name_or_path is None:
+        model = UNet2DModel(
+                sample_size= args.resolution,
+                in_channels=3,
+                out_channels=3,
+                layers_per_block=2,
+                num_class_embeds=1000,
+                block_out_channels= [32, 64],
+                attention_head_dim=8,
+                down_block_types= [
+                    "ResnetDownsampleBlock2D",
+                    "AttnDownBlock2D",
+                ],
+                up_block_types= [
+                    "AttnUpBlock2D",
+                    "ResnetUpsampleBlock2D",
+                ],
+                resnet_time_scale_shift="scale_shift",
+                upsample_type="resnet",
+                downsample_type="resnet"
+                )
+        target_model = UNet2DModel(
+                sample_size= args.resolution,
+                in_channels=3,
+                out_channels=3,
+                layers_per_block=2,
+                num_class_embeds=1000,
+                block_out_channels= [32, 64],
+                attention_head_dim=8,
+                down_block_types= [
+                    "ResnetDownsampleBlock2D",
+                    "AttnDownBlock2D",
+                ],
+                up_block_types= [
+                    "AttnUpBlock2D",
+                    "ResnetUpsampleBlock2D",
+                ],
+                resnet_time_scale_shift="scale_shift",
+                upsample_type="resnet",
+                downsample_type="resnet"
+                )
+    else:
+        config = UNet2DModel.load_config(args.model_config_name_or_path)
+        model = UNet2DModel.from_config(config)
+        target_model = UNet2DModel.from_config(config)
```

Comment:
So, this, in a sense, is step distillation i.e., we're not essentially compressing the model but reducing the number of steps.

Reply:
Perhaps it is a distinction without a difference, but I think of it as reducing the number of steps for a (diffusion model, sampler) pair. In my mind, this distinction is important because my guess is that the quality of the sampler affects the quality of the distilled consistency model; if we distill using a sampler which produces bad samples for the diffusion model, the consistency model might learn to faithfully produce similar samples to sampling from the diffusion model using the sampler, but which are still poor quality because the samples it learned from are poor quality. (Conversely, using a better sampler might produce a better consistency model.)